### ISSUE TITLE
Frozen bytecode compilation fails on strings containing double quotes

### ISSUE THREAD

**Issue by ryannathans** [micropython/micropython#2104]
I have a file called that contains the following

```
foo = "\"bar\""
```

and another that contains the following (single quote, double quote, double quote, single quote)

```
foo = '"bar"'
```

both throw errors during frozen bytecode compilation, it seems that the parser treats the quotes incorrectly.

```
build-STM32F439/scripts/frozen_mpy.c:1320:39: error: expected '}' before 'bar'
         (const byte*)"\xd4\xe5\x05" ""bar"",
```

It's strange, because the following works (double quote, single quote, single quote, double quote)

```
foo = "'bar'"
```

**Comment by ryannathans** [comment]
foo = """\"bar\""""

also fails compilation. it seems as though if any string contains " in the final interpreted string, the frozen compilation breaks

**Comment by dpgeorge** [comment]
Should be fixed by 202d5acd06d320f3d2485b85d4ff6c412e50ad21.

**Comment by ryannathans** [comment]
Confirmed, fixed.
